lib/ansible/modules/cloud/amazon/ec2_vpc_nacl.py

```diff
@@ -297,13 +293,20 @@ def rules_changed(aws_rules, param_rules, Egress, nacl_id, client, module):
     return changed
 
 
-def process_rule_entry(entry, Egress):
+def process_rule_entry(entry, Egress, module):
     params = dict()
     params['RuleNumber'] = entry[0]
     params['Protocol'] = str(PROTOCOL_NUMBERS[entry[1]])
```

Comment:
I know this isn't part of your changes but I just noticed that this will result in an unhandled KeyError exception if the user specifies a protocol number that isn't in the `PROTOCOL_NUMBERS` constant. Technically any valid protocol number is allowed but anything other than `-1 (all), 1 (icmp), 6 (tcp), 17 (udp), 58 (ipv6-icmp)` will result in the rule ignoring the port range or icmp type/code values.

here are the confusing rules from boto3 for the protocol handling.

> The protocol number. A value of "-1" means all protocols. If you specify "-1" or a protocol number other than "6" (TCP), "17" (UDP), or "1" (ICMP), traffic on all ports is allowed, regardless of any ports or ICMP types or codes that you specify. If you specify protocol "58" (ICMPv6) and specify an IPv4 CIDR block, traffic for all ICMP types and codes allowed, regardless of any that you specify. If you specify protocol "58" (ICMPv6) and specify an IPv6 CIDR block, you must specify an ICMP type and code.

Reply:
Committed ab80ad07c163ff4a23a3f203140612e4e4da0bd6 to support ipv6-icmp.